core/templates/pages/exploration-editor-page/feedback-tab/services/thread-data-backend-api.service.ts

```diff
@@ -0,0 +1,306 @@
+    private suggestionsService: SuggestionsService,
+    private threadMessageObjectFactory: ThreadMessageObjectFactory,
+    private urlInterpolationService: UrlInterpolationService
+  ) {}
+
+  getFeedbackStatsHandlerUrl(): string {
+    return this.urlInterpolationService.interpolateUrl(
+      '/feedbackstatshandler/<exploration_id>', {
+        exploration_id: this.contextService.getExplorationId()
+      });
+  }
+
+  getFeedbackThreadViewEventUrl(threadId: string): string {
+    return this.urlInterpolationService.interpolateUrl(
+      '/feedbackhandler/thread_view_event/<thread_id>', {
+        thread_id: threadId
+      });
+  }
+
+  getSuggestionActionHandlerUrl(threadId: string): string {
+    return this.urlInterpolationService.interpolateUrl(
+      '/suggestionactionhandler/exploration/<exploration_id>/<thread_id>', {
+        exploration_id: this.contextService.getExplorationId(),
+        thread_id: threadId
+      });
+  }
+
+  getSuggestionListHandlerUrl(): string {
+    return '/suggestionlisthandler';
+  }
+
+  getThreadHandlerUrl(threadId: string): string {
+    return this.urlInterpolationService.interpolateUrl(
+      '/threadhandler/<thread_id>', {
+        thread_id: threadId
+      });
+  }
+
+  getThreadListHandlerUrl(): string {
+    return this.urlInterpolationService.interpolateUrl(
+      '/threadlisthandler/<exploration_id>', {
+        exploration_id: this.contextService.getExplorationId()
+      });
+  }
+
+  setFeedbackThreadFromBackendDict(
+      threadBackendDict: FeedbackThreadBackendDict): FeedbackThread {
+    if (!threadBackendDict) {
+      throw new Error('Missing input backend dict');
+    }
+    let thread = this.feedbackThreadObjectFactory.createFromBackendDict(
+      threadBackendDict);
+    this.threadsById.set(thread.threadId, thread);
+    return thread;
+  }
+
+  setSuggestionThreadFromBackendDicts(
+      threadBackendDict: FeedbackThreadBackendDict,
+      suggestionBackendDict: SuggestionBackendDict): SuggestionThread {
+    if (!threadBackendDict || !suggestionBackendDict) {
+      throw new Error('Missing input backend dicts');
+    }
+    let thread = this.suggestionThreadObjectFactory.createFromBackendDicts(
+      threadBackendDict, suggestionBackendDict);
+    this.threadsById.set(thread.threadId, thread);
+    return thread;
+  }
+
+  getThread(threadId: string): AnyThread {
+    return this.threadsById.get(threadId) || null;
+  }
+
+  getThreadsAsync(): Promise<SuggestionAndFeedbackThreads> {
+    // TODO(#8016): Move this http call to a backend-api.service with unit
+    // tests.
+    let suggestionsPromise = this.http.get(this.getSuggestionListHandlerUrl(), {
+      params: {
+        target_type: 'exploration',
+        target_id: this.contextService.getExplorationId()
+      }
+    });
+    // TODO(#8016): Move this http call to a backend-api.service with unit
+    // tests.
+    let threadsPromise = this.http.get(this.getThreadListHandlerUrl());
```

Comment:
The name should be threads$ since it is an Observable.

Reply:
Done